x-pack/platform/plugins/shared/cases/public/components/all_cases/selector_modal/use_cases_add_to_existing_case_modal.tsx

```diff
@@ -137,11 +137,23 @@ export const useCasesAddToExistingCaseModal = ({
     }: {
       getAttachments?: ({ theCase }: { theCase?: CaseUI }) => CaseAttachmentsWithoutOwner;
     } = {}) => {
+      // Extract alert IDs from attachments to show indicators in the modal
+      const attachments = getAttachments?.({ theCase: undefined }) ?? [];
+      const alertIds = attachments
+        .filter((attachment) => attachment.type === 'alert')
+        .flatMap((attachment) => {
+          if (attachment.alertId) {
+            return Array.isArray(attachment.alertId) ? attachment.alertId : [attachment.alertId];
+          }
+          return [];
+        });
```

Comment:
The typing was a bit broken here. We can inline a type guard to the `filter` call and it makes it play nicer. Also avoided nesting in the `flatMap`.

```suggestion
      const alertIds = attachments
        .filter((attachment): attachment is Extract<typeof attachment, { type: 'alert' }> => 
          attachment.type === 'alert'
        )
        .flatMap((attachment) => {
          const { alertId } = attachment;
          return alertId ? (Array.isArray(alertId) ? alertId : [alertId]) : [];
        });
```